### PR TITLE
Fix Bool pattern matching to be exhaustive

### PR DESCRIPTION
**Pull request by lolgab**
Fixes https://github.com/com-lihaoyi/upickle/issues/456

### CODE REVIEW

File: upack/test/src/upack/BoolTests.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          case upack.Bool(value)
+            if value == bool
+            && jsb.value == value =>
+        }
+      }
+    }
+    test("upack.Bool pattern matching should be exhaustive") {
+      upack.Bool(true) match {
+        case upack.Bool(value) =>
+      }
```
Comment (lefou) [inline upack/test/src/upack/BoolTests.scala:29]: This test assumes we're using `-Xfatal-warnings`, right? If not, I don't see how this test is supposed to detect non-exhaustiveness.

Comment (lolgab) [reply upack/test/src/upack/BoolTests.scala:29]: Yes, exactly. I added it in the previous PR exactly for this test. I don't know if there is a better way to test this.

Comment (lefou) [reply upack/test/src/upack/BoolTests.scala:29]: That means, if the settings get changed later, we have no indicator that this test is insufficient. The bare minimum is a comment in code or in the test name. A bit better would be to detect the flags (e.g. by generating some build info code). I know utest has `compileError`; maybe it also fails a warning under `-Xfatal-warnings`, in which case we could guard some code that emits a warning with `compileError` to detect when this setting gets removed in the future.

**Comment by lihaoyi** [comment]
Gonna merge this to get it in for the 3.0.0 release

**Pull request merged**